api/src/main/java/com/flipkart/poseidon/api/APIApplication.java

```diff
@@ -51,7 +53,22 @@ public void init(ExecutorService datasourceTPE, ExecutorService filterTPE) {
 
     @Override
     public void handleRequest(PoseidonRequest request, PoseidonResponse response) throws ElementNotFoundException, BadRequestException, ProcessingException, InternalErrorException {
-        lego.buildResponse(request, response);
+        HttpMethod method = (HttpMethod) request.getAttribute(RequestConstants.METHOD);
+        boolean handled = false;
+        if (method != null && method.equals(HttpMethod.OPTIONS)) {
+            handled = handleOptionsRequest(request, response);
+        }
+
+        if (!handled) {
+            lego.buildResponse(request, response);
+        }
+    }
+
+    /**
+     * return true if providing a custom implementation for OPTION calls
+     */
+    public boolean handleOptionsRequest(PoseidonRequest req, PoseidonResponse resp) {
```

Comment:
Let this method throw same exceptions like handleRequest()
throws ElementNotFoundException, BadRequestException, ProcessingException, InternalErrorException